compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp

```diff
@@ -111,12 +114,8 @@ LogicalResult assignNpuDmaBdIds(AMDAIE::WorkgroupOp workgroupOp) {
           return WalkResult::interrupt();
         ChannelBdIdGenerator &generator =
             shimTileToGeneratorMap[tileOp.getResult()];
-        // TODO(jornt): Temporarily use channel 0 for all DMAs. This should
-        // return correct results for Shim channels, however, for generality
-        // towards other DMAs and future hardware generations, channel
-        // assignment should happen before BD assignemnt. This requires more
-        // refactoring.
-        std::optional<uint32_t> bdId = generator.getAndAssignBdId(0);
+        std::optional<uint32_t> bdId = generator.getAndAssignBdId(
+            channel, BdIdAssignmentMode::Incremental);
```

Comment:
> During the AMDAIEAssignNpuDmaBdIds pass, I am currently not getting rid of any redundant dma_wait

Right, yeah, I see. Otherwise, `0` will always be the smallest available `bd_id`.